### PR TITLE
Add integration specs & fix N+1 problems

### PR DESCRIPTION
**Pull request by Gopxfs**
# Summary 📒 
For this milestone, we:
- Added integration specs to:
  - Users list (main page).
  - Each user profile.
  - User's list of posts.
  - Each user's individual post.
- Fixed N+1 problems on posts #show & posts #index.
- Added bullet & database_cleaner gems.

### CODE REVIEW

File: spec/requests/posts_controller_spec.rb [hunk 1]
```diff
@@ -1,27 +0,0 @@
-require 'rails_helper'
-
-RSpec.describe 'Posts Controller', type: :request do
-  describe 'GET #index' do
```
Comment (Munsa1) [inline spec/requests/posts_controller_spec.rb:4]: - [ ] You have failing tests, kindly fix them.
image

image

image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Gopxfs, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_controller_spec.rb (hunk 1 above)
Comment (Gopxfs) [reply spec/requests/posts_controller_spec.rb:4]: Hey @Munsa1 , this folder has been removed, I think you pulled the wrong branch

**Review by Wes-Isaac — APPROVED** [review]
Hi @Gopxfs, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Professional README 👀.
- Great PR description 👍.
- No linter error ✅.


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**